src/theory/strings/theory_strings.cpp

```diff
@@ -749,10 +749,62 @@ void TheoryStrings::eqNotifyNewClass(TNode t){
     Trace("strings-debug") << "New length eqc : " << t << std::endl;
     //we care about the length of this string
     d_termReg.registerTerm(t[0], 1);
+
+    eq::EqualityEngine* ee = d_state.getEqualityEngine();
+    Node r = ee->getRepresentative(t[0]);
+    EqcInfo* ei = d_state.getOrMakeEqcInfo(r);
+    if (k == STRING_LENGTH)
+    {
+      ei->d_lengthTerm = t;
+    }
+    else
+    {
+      ei->d_codeTerm = t[0];
+    }
   }
   d_eagerSolver.eqNotifyNewClass(t);
 }
 
+void TheoryStrings::eqNotifyMerge(TNode t1, TNode t2)
+{
+  d_eagerSolver.eqNotifyMerge(t1, t2);
```

Comment:
Can you move this to line 777 (or maybe to line 779 even)?  Also it might be worth considering passing e1, e2 to this method to avoid extra lookups.

Reply:
Yes, that makes sense. I have added `e1` and `e2` as arguments to `eqNotifyMerge`.